cpp/unit/unit.h

```diff
@@ -43,9 +43,9 @@ class Unit : public log::LogSource {
 	const id_t id;
 
 	/**
-	 * producer used to create this object
+	 * type which was used to create this object
```

Comment:
is the type really used to "create" the object? can't a unit's type actually change on the fly?